cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/mtom/MtomResponseWrapper.java

```diff
@@ -0,0 +1,159 @@
+package nl.nn.adapterframework.extensions.cmis.mtom;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import javax.mail.BodyPart;
+import javax.mail.internet.MimeMultipart;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.InputStreamEntity;
+import org.apache.http.entity.mime.FormBodyPartBuilder;
+import org.apache.http.entity.mime.content.InputStreamBody;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.logging.log4j.Logger;
+import org.springframework.mock.web.DelegatingServletOutputStream;
+
+import nl.nn.adapterframework.http.InputStreamDataSource;
+import nl.nn.adapterframework.http.mime.MultipartEntityBuilder;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class MtomResponseWrapper extends HttpServletResponseWrapper {
+	protected Logger log = LogUtil.getLogger(this);
+
+	private ContentType contentType;
+
+	public MtomResponseWrapper(ServletResponse response) throws IOException {
+		this((HttpServletResponse) response);
+	}
+
+	public MtomResponseWrapper(HttpServletResponse response) throws IOException {
+		super(response);
+	}
+
+	@Override
+	public ServletOutputStream getOutputStream() throws IOException {
+
+		contentType = ContentType.parse(getContentType());
+		if(log.isTraceEnabled()) log.trace("recieved response with ContentType ["+contentType+"]");
+
+		// Als mimeType == text/html dan geen multipart doen :)
+		if(!contentType.getMimeType().contains("multipart")) {
+			return super.getOutputStream();
+		} else {
+			return new DelegatingServletOutputStream(new MtomOutputStream(super.getOutputStream()));
+		}
+	}
+
+	private class MtomOutputStream extends FilterOutputStream {
+		private ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();
+
+		public MtomOutputStream(OutputStream out) {
+			super(out);
+		}
+
+		@Override
+		public void write(int b) throws IOException {
+			bufferStream.write(b);
+		}
+
+		@Override
+		public void write(byte[] b, int off, int len) throws IOException {
+			bufferStream.write(b, off, len);
+		}
+
+		@Override
+		public synchronized void flush() throws IOException {
+			try {
+				ByteArrayInputStream is = new ByteArrayInputStream(bufferStream.toByteArray());
+				InputStreamDataSource dataSource = new InputStreamDataSource(contentType.toString(), is);
```

Comment:
Hier lees je de hele stream in een byte array in geheugen in, om hem daarna maar één keer als stream te lezen

Reply:
Ja klopt.